app/src/main/java/org/wikipedia/feed/onthisday/OnThisDayFragment.java

```diff
@@ -310,63 +315,78 @@ public int getItemViewType(int position) {
             yearTextView.setText(DateUtil.yearToStringWithEra(event.year()));
             yearsInfoTextView.setText(DateUtil.getYearDifferenceString(event.year()));
-            if (prevEvent != null && prevEvent.year() == event.year()) {
-                yearContainer.setVisibility(View.GONE);
-                yearSpace.setVisibility(View.GONE);
-            } else {
-                yearContainer.setVisibility(View.VISIBLE);
-                yearSpace.setVisibility(prevEvent == null ? View.GONE : View.VISIBLE);
-            }
-        }
-
-        private void setPads() {
-            int pad1 = (int) DimenUtil.dpToPx(PADDING1);
-            int pad2 = (int) DimenUtil.dpToPx(PADDING2);
-            int pad3 = (int) DimenUtil.dpToPx(PADDING3);
-
-            descTextView.setPaddingRelative(pad1, 0, 0, 0);
-            pagesRecycler.setPaddingRelative(pad2, 0, 0, 0);
-            yearTextView.setPaddingRelative(pad3, 0, 0, 0);
+            setPagesViewPager(event);
         }
 
-        private void setPagesRecycler(OnThisDay.Event event) {
+        private void setPagesViewPager(OnThisDay.Event event) {
             if (event.pages() != null) {
-                OnThisDayCardView.RecyclerAdapter recyclerAdapter = new OnThisDayCardView.RecyclerAdapter(getChildFragmentManager(), event.pages(), wiki, false);
-                pagesRecycler.setAdapter(recyclerAdapter);
+                ViewPagerAdapter viewPagerAdapter = new ViewPagerAdapter(getChildFragmentManager(), event.pages(), wiki);
+                pagesViewPager.setAdapter(viewPagerAdapter);
+                pagesViewPager.setOffscreenPageLimit(2);
+                 new TabLayoutMediator(pagesIndicator, pagesViewPager, (tab, position) -> { }).attach();
+                pagesViewPager.setVisibility(VISIBLE);
+                pagesIndicator.setVisibility(event.pages().size() == 1 ? GONE : VISIBLE);
             } else {
-                pagesRecycler.setVisibility(View.GONE);
+                pagesViewPager.setVisibility(GONE);
+                pagesIndicator.setVisibility(GONE);
             }
         }
+
+        public void animateRadioButton() {
+            Animation pulse = AnimationUtils.loadAnimation(getContext(), R.anim.pulse);
+            radioButtonImageView.startAnimation(pulse);
+        }
     }
 
     private class FooterViewHolder extends RecyclerView.ViewHolder {
         FooterViewHolder(View v) {
             super(v);
-            View backToTopView = v.findViewById(R.id.back_to_top_view);
-            backToTopView.setOnClickListener(v1 -> {
+            View backToFutureView = v.findViewById(R.id.back_to_future_text_view);
+            backToFutureView.setOnClickListener(v1 -> {
                 appBarLayout.setExpanded(true);
                 eventsRecycler.scrollToPosition(0);
             });
         }
     }
+     class ViewPagerAdapter extends RecyclerView.Adapter<OnThisDayPagesViewHolder> {
```

Comment:
Add `static`?

Reply:
So ..the message that Android studio is annoying us with is because non-static inner classes hold a direct reference to outer class... and since some inner classes can live longer than the container, if the activity or fragment holding it gets destroyed, then it will not be garbage collected until this class is done with its stuff. Hence the memory leak. Here it is not an issue as we are sure it is tied to the Fragment's lifecycle. However, I think we need to take a call on whether or not we should start making them all static to get rid or warnings, or play it by the ear.. because there are other places where we have just left this... 😅